pytorch_lightning/trainer/trainer.py

```diff
@@ -747,6 +744,9 @@ def run_evaluation(self, on_epoch=False):
 
         torch.set_grad_enabled(True)
 
+        # reset cached results
+        self.logger_connector.reset()
+
```

Comment:
nit: why not move this to right after L737?

Reply:
changed!